src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.OSX.cs

```diff
@@ -18,10 +18,12 @@ private bool InitNativeCultureDataCore()
             string realNameBuffer = _sRealName;
 
             _sWindowsName = GetLocaleNameNative(realNameBuffer);
+            _sName = _sWindowsName;
+            _sRealName = _sName;
             return true;
         }
 
-        internal static unsafe string GetLocaleNameNative(string localeName)
+        internal static string GetLocaleNameNative(string localeName)
```

Comment:
Functions where we interop with C should be marked with `unsafe`. In more than one place it got removed.